[Automation] native-image run fails for org.springframework.boot:spring-boot-configuration-processor:2.7.18

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581584
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
adata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.springframework.boot/spring-boot-configuration-processor/2.7.18 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AllowJRTFileSystem@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=5c09f739-2487-8970-d5e7-3ecf10d41713 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (5.8s @ 0.16GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
Warning: Unknown attribute(s) [module] in bundle descriptor object
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (43.8s @ 1.90GB)
   11,665 types,  15,243 fields, and  55,428 methods found reachable
    3,744 types,     307 fields, and   2,242 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (6.4s @ 1.24GB)
[4/8] Parsing methods...      [**]                                                                       (4.7s @ 1.35GB)
[5/8] Inlining methods...     [***]                                                                      (2.8s @ 1.43GB)
[6/8] Compiling methods...    [*****]                                                                   (26.4s @ 1.02GB)
[7/8] Laying out methods...   [**]                                                                       (4.2s @ 1.33GB)
[8/8] Creating image...       [**]                                                                       (4.6s @ 1.50GB)
  16.80MB (39.44%) for code area:    35,767 compilation units
  20.19MB (47.39%) for image heap:  308,692 objects and 169 resources
   5.61MB (13.17%) for other data
  42.59MB in total image size, 39.52MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   5.80MB java.base                                            5.21MB byte[] for code metadata
   4.55MB jdk.compiler                                         3.72MB byte[] for java.lang.String
   2.96MB svm.jar (Native Image)                               2.03MB java.lang.String
 566.10kB assertj-core-3.22.0.jar                              1.58MB java.lang.Class
 481.21kB java.xml                                             1.21MB com.oracle.svm.core.hub.DynamicHubCompanion
 240.45kB jdk.javadoc                                        799.78kB byte[] for general heap data
 208.61kB junit-jupiter-engine-5.9.2.jar                     550.86kB heap alignment
 151.09kB jdk.zipfs                                          469.02kB java.util.concurrent.ConcurrentHashMap$Node
 121.70kB junit-platform-launcher-1.9.2.jar                  445.22kB java.util.HashMap$Node
 120.93kB spring-boot-configuration-processor-2.7.18.jar     357.78kB java.lang.Object[]
   1.03MB for 30 more packages                                 3.81MB for 2776 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 19 component(s) is embedded in binary (999B). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.6s (8.4% of total time) in 1036 GCs | Peak RSS: 2.76GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' in 1m 41s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest > javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors() FAILED

org_springframework_boot.spring_boot_configuration_processor.TreesTest > javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment() FAILED

org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest > javaCompilerFieldValueParsingVisitsVariableTreeAccessors() FAILED


Failures (3):
  JUnit Jupiter:ExpressionTreeTest:javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.3/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.3/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.Locations.<clinit>(Locations.java:139)
       [...]
  JUnit Jupiter:TreesTest:javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.TreesTest', methodName = 'javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.TreesTest.javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment(TreesTest.java:53)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:VariableTreeTest:javaCompilerFieldValueParsingVisitsVariableTreeAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsVariableTreeAccessors', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.3/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.3/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest.javaCompilerFieldValueParsingVisitsVariableTreeAccessors(VariableTreeTest.java:47)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 4 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         3 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 45s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-configuration-processor:2.7.18 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 47s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.7.18 at 2026-05-12T10:58:27Z with exit code 1 after 108s
❌ ERROR: 'native-image run' for 2.7.18 failed with exit code 1.
FAILED[native-image run][2.7.18][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581548
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
r-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.springframework.boot/spring-boot-configuration-processor/2.7.18 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AllowJRTFileSystem@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=7c159a96-c097-5a6e-0c3e-f68c11ea3431 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.8s @ 0.17GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (48.1s @ 1.79GiB)
   11,826 types,  15,327 fields, and  55,993 methods found reachable
      429 types,      98 fields, and   1,107 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      169 resource accesses registered with 262.89KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.9s @ 1.20GiB)
[4/8] Parsing methods...      [***]                                                                     (5.3s @ 1.29GiB)
[5/8] Inlining methods...     [***]                                                                     (2.7s @ 1.37GiB)
[6/8] Compiling methods...    [*****]                                                                  (28.5s @ 0.99GiB)
[7/8] Laying out methods...   [**]                                                                      (4.3s @ 1.28GiB)
[8/8] Creating image...       [**]                                                                      (4.8s @ 1.45GiB)
  14.61MiB (36.79%) for code area:    36,135 compilation units
  19.63MiB (49.42%) for image heap:  321,144 objects and 30 resources
   5.48MiB (13.80%) for other data
  39.71MiB in total image size, 36.75MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.79MiB jdk.compiler/com.sun.tools.javac.comp               4.84MiB byte[] for code metadata
   1.40MiB o.g.n.runtime/com.oracle.svm.core.code              3.67MiB byte[] for string data
 759.67KiB java.base/java.util                                 2.02MiB java.base/java.lang.String
 477.48KiB jdk.compiler/com.sun.tools.javac.code               1.52MiB java.base/java.lang.Class
 445.08KiB jdk.compiler/com.sun.tools.javac.jvm                1.26MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 408.32KiB java.base/jdk.internal.classfile.impl             979.67KiB byte[] for general heap data
 322.07KiB java.base/java.lang                               459.45KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 289.86KiB jdk.compiler/com.sun.tools.javac.parser           427.24KiB byte[] for reflection metadata
 284.82KiB jdk.compiler/com.sun.tools.javac.tree             407.81KiB java.base/java.util.HashMap$Node
 266.80KiB java.base/com.sun.crypto.provider                 374.88KiB java.base/java.lang.Object[]
   8.25MiB for 364 more packages                               3.72MiB for 2793 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.04KiB). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.7s (7.8% of total time) in 1055 GCs | Peak RSS: 2.56GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests (executable, 36.75MiB)
 /tmp/native-image-build-output.json (build_info, 1.07KiB)
========================================================================================================================
Finished generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' in 1m 50s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest > javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors() FAILED

org_springframework_boot.spring_boot_configuration_processor.TreesTest > javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment() FAILED

org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest > javaCompilerFieldValueParsingVisitsVariableTreeAccessors() FAILED


Failures (3):
  JUnit Jupiter:ExpressionTreeTest:javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.Locations.<clinit>(Locations.java:139)
       [...]
  JUnit Jupiter:TreesTest:javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.TreesTest', methodName = 'javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.TreesTest.javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment(TreesTest.java:53)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:VariableTreeTest:javaCompilerFieldValueParsingVisitsVariableTreeAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsVariableTreeAccessors', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest.javaCompilerFieldValueParsingVisitsVariableTreeAccessors(VariableTreeTest.java:47)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 3 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         3 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 54s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-configuration-processor:2.7.18 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 56s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.7.18 at 2026-05-12T11:03:12Z with exit code 1 after 116s
❌ ERROR: 'native-image run' for 2.7.18 failed with exit code 1.
FAILED[native-image run][2.7.18][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581557
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
upiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.springframework.boot/spring-boot-configuration-processor/2.7.18 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:+AllowJRTFileSystem@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=00a77bc4-cf8f-a684-b755-f24ac1bf7499 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.2s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (42.0s @ 1.84GiB)
   11,885 types,  15,479 fields, and  56,587 methods found reachable
      447 types,      98 fields, and   1,146 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      170 resource accesses registered with 263.00KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.5s @ 1.22GiB)
[4/8] Parsing methods...      [**]                                                                      (4.3s @ 1.31GiB)
[5/8] Inlining methods...     [***]                                                                     (2.3s @ 1.39GiB)
[6/8] Compiling methods...    [*****]                                                                  (25.6s @ 1.00GiB)
[7/8] Laying out methods...   [**]                                                                      (4.3s @ 1.29GiB)
[8/8] Creating image...       [**]                                                                      (4.6s @ 1.49GiB)
  14.82MiB (36.59%) for code area:    36,627 compilation units
  20.19MiB (49.84%) for image heap:  325,273 objects and 31 resources
   5.50MiB (13.57%) for other data
  40.50MiB in total image size, 37.50MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.79MiB jdk.compiler/com.sun.tools.javac.comp               4.91MiB byte[] for code metadata
   1.42MiB o.g.n.runtime/com.oracle.svm.core.code              3.70MiB byte[] for string data
 761.41KiB java.base/java.util                                 2.02MiB java.base/java.lang.String
 477.48KiB jdk.compiler/com.sun.tools.javac.code               1.53MiB java.base/java.lang.Class
 445.08KiB jdk.compiler/com.sun.tools.javac.jvm                1.27MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 408.32KiB java.base/jdk.internal.classfile.impl             974.38KiB byte[] for general heap data
 327.47KiB java.base/java.lang                               551.89KiB heap alignment
 289.86KiB jdk.compiler/com.sun.tools.javac.parser           461.76KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 284.86KiB jdk.compiler/com.sun.tools.javac.tree             444.71KiB byte[] for reflection metadata
 267.01KiB java.base/com.sun.crypto.provider                 404.75KiB java.base/java.util.HashMap$Node
   8.42MiB for 367 more packages                               3.99MiB for 2788 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (1.05KiB). 10 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.9s (8.9% of total time) in 1004 GCs | Peak RSS: 2.61GiB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests (executable, 37.50MiB)
 /tmp/native-image-build-output.json (build_info, 1.07KiB)
========================================================================================================================
Finished generating 'org.springframework.boot.spring-boot-configuration-processor_tests-tests' in 1m 38s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest > javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors() FAILED

org_springframework_boot.spring_boot_configuration_processor.TreesTest > javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment() FAILED

org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest > javaCompilerFieldValueParsingVisitsVariableTreeAccessors() FAILED


Failures (3):
  JUnit Jupiter:ExpressionTreeTest:javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.ExpressionTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsExpressionTreeFactoryAndArrayAccessors', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.Locations.<clinit>(Locations.java:139)
       [...]
  JUnit Jupiter:TreesTest:javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.TreesTest', methodName = 'javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.TreesTest.javaCompilerFieldValuesParserUnwrapsDelegatingProcessingEnvironment(TreesTest.java:53)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:VariableTreeTest:javaCompilerFieldValueParsingVisitsVariableTreeAccessors()
    MethodSource [className = 'org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest', methodName = 'javaCompilerFieldValueParsingVisitsVariableTreeAccessors', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class com.sun.tools.javac.file.Locations
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.createLocations(BaseFileManager.java:126)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.BaseFileManager.<init>(BaseFileManager.java:82)
       jdk.compiler@25.0.2/com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:161)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:108)
       jdk.compiler@25.0.2/com.sun.tools.javac.api.JavacTool.getStandardFileManager(JavacTool.java:68)
       org_springframework_boot.spring_boot_configuration_processor.VariableTreeTest.javaCompilerFieldValueParsingVisitsVariableTreeAccessors(VariableTreeTest.java:47)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 4 ms
[         5 containers found      ]
[         0 containers skipped    ]
[         5 containers started    ]
[         0 containers aborted    ]
[         5 containers successful ]
[         0 containers failed     ]
[         3 tests found           ]
[         0 tests skipped         ]
[         3 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         3 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build/native/nativeTestCompile/org.springframework.boot.spring-boot-configuration-processor_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 42s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-configuration-processor:2.7.18 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 44s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.7.18 at 2026-05-12T11:00:35Z with exit code 1 after 105s
❌ ERROR: 'native-image run' for 2.7.18 failed with exit code 1.
FAILED[native-image run][2.7.18][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.7.18" ./gradlew clean test -Pcoordinates="org.springframework.boot:spring-boot-configuration-processor:2.7.18"]
```

--- a/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build.gradle
+++ b/tests/src/org.springframework.boot/spring-boot-configuration-processor/2.7.18/build.gradle
@@ -4,11 +4,18 @@
  * You should have received a copy of the CC0 legalcode along with this
  * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
  */
+import org.graalvm.buildtools.gradle.tasks.NativeRunTask
+
 plugins {
     id "org.graalvm.internal.tck"
 }
 
 String libraryVersion = tck.testedLibraryVersion.get()
+String javaHome = System.getenv('JAVA_HOME')
+List<String> javacRuntimeArgs = [
+        "-Djava.home=${javaHome}",
+        "-Djava.class.path=${-> sourceSets.test.runtimeClasspath.asPath}"
+]
 
 dependencies {
     testImplementation "org.springframework.boot:spring-boot-configuration-processor:$libraryVersion"
@@ -19,8 +26,7 @@ graalvmNative {
     binaries {
         test {
             buildArgs("-H:+AllowJRTFileSystem")
-            runtimeArgs("-Djava.home=${System.getenv('JAVA_HOME')}")
-            runtimeArgs("-Djava.class.path=${sourceSets.test.runtimeClasspath.asPath}")
+            runtimeArgs.addAll(javacRuntimeArgs)
         }
     }
     agent {
@@ -31,4 +37,8 @@ graalvmNative {
             }
         }
     }
+}
+
+tasks.named("nativeTest", NativeRunTask) {
+    runtimeArgs.addAll(javacRuntimeArgs)
 }

